lib/AboutPreferences.jsm

```diff
@@ -257,6 +260,36 @@ this.AboutPreferences = class AboutPreferences {
       });
     });
 
+    try {
+      const discoveryStreamPref = JSON.parse(ASPreferences.values["discoverystream.config"]);
+      if (discoveryStreamPref.enabled) {
+        // If Discovery Stream is enabled hide Home Content options
+        contentsGroup.style.visibility = "hidden";
+
+        const discoveryGroup = homeGroup.insertAdjacentElement("afterend", homeGroup.cloneNode());
+        discoveryGroup.id = "discoveryContentsGroup";
+        discoveryGroup.setAttribute("data-subcategory", "contents");
```

Comment:
Probably make the subcategory "discovery". As this is now, it prevents `about:preferences#home-contents` from working when it's not enabled (as it tries to highlight a hidden section).